romancal/multiband_catalog/multiband_catalog.py

```diff
@@ -318,3 +323,77 @@ def make_source_injected_library(library):
 
     # Return library of source injected models and injection catalog
     return ModelLibrary(si_model_lst), si_cat
+
+
+def match_recovered_sources(original, injected, si_catalogs):
+    """
+    Create recovered sources which matches sources between
+    an original catalog and injected catalog.
+
+    Parameters
+    -----------
+    original : `~astropy.table.Table`
+        Catalog of original sources.
+
+    injected : `~astropy.table.Table`
+        Catalog of injected sources.
+
+    si_catalogs : `~astropy.table.Table`
+        Catalog of sources after injection.
+
+    Returns
+    -------
+    recovered : `~astropy.table.Table`
+        Catalog of recovered sources.
+    """
+
+    # Create recovered catalog
+    recovered = copy.deepcopy(si_catalogs)
+    recovered["best_injected_index"] = -1
+
+    # Create skycoord objects
+    if hasattr(original["ra"], "unit") and (original["ra"].unit == u.deg):
+        ocoord = coordinates.SkyCoord(original["ra"], original["dec"])
+    else:
+        ocoord = coordinates.SkyCoord(original["ra"] * u.deg, original["dec"] * u.deg)
+
+    if hasattr(injected["ra"], "unit") and (injected["ra"].unit == u.deg):
+        icoord = coordinates.SkyCoord(injected["ra"], injected["dec"])
+    else:
+        icoord = coordinates.SkyCoord(injected["ra"] * u.deg, injected["dec"] * u.deg)
+
+    if hasattr(recovered["ra"], "unit") and (recovered["ra"].unit == u.deg):
+        rcoord = coordinates.SkyCoord(recovered["ra"], recovered["dec"])
+    else:
+        rcoord = coordinates.SkyCoord(recovered["ra"] * u.deg, recovered["dec"] * u.deg)
+
+    # Set maximum radius for matching.
+    # d = 3*sqrt(half_right_radius^2 + 0.2^2)
+    # In addition, set a maximum of 10"
+    max_sep_injected = (
+        3 * np.sqrt(injected["half_light_radius"].value ** 2 + 0.2**2) * u.arcsec
+    )
+    sep_mask = np.greater(max_sep_injected.value, 10)
```

Comment:
```suggestion
    sep_mask = np.greater(max_sep_injected.to(u.arcsec).value, 10)
```
or np.greater(max_sep_injected, 10 * u.arcsec)?